itest/psbt_test.go

```diff
@@ -1608,6 +1619,436 @@ func testPsbtSighashNoneInvalid(t *harnessTest) {
+	require.Nil(t.t, vPkt.Outputs[0].Asset.PrevWitnesses[0].SplitCommitment)
+
+	fundedPsbtBytes, err := tappsbt.Encode(vPkt)
+	require.NoError(t.t, err)
+
+	// Alice signs the vPSBT.
+	signedResp, err := alice.SignVirtualPsbt(
+		ctxb, &wrpc.SignVirtualPsbtRequest{
+			FundedPsbt: fundedPsbtBytes,
+		},
+	)
+	require.NoError(t.t, err)
+	require.Contains(t.t, signedResp.SignedInputs, uint32(0))
+
+	// Deserialize the signed vPSBT.
+	vPkt, err = tappsbt.Decode(signedResp.SignedPsbt)
+	require.NoError(t.t, err)
+
+	// Now we need to create the bitcoin PSBT where the previously created
+	// vPSBT will be anchored to.
+	btcpsbt, err := tapsend.PrepareAnchoringTemplate([]*tappsbt.VPacket{
+		vPkt,
+	})
+	require.NoError(t.t, err)
+
+	// This bitcoin PSBT should have 1 input, which is the anchor of Alice's
+	// assets, and 2 outputs that correspond to Alice's anchor that carries the
+	// transfered assets, and one output that carries the bitcoin change.
+	require.Len(t.t, btcpsbt.Inputs, 1)
+	require.Len(t.t, btcpsbt.Outputs, 2)
+
+	// Let's set an actual address for Alice's output.
+	addrResp := t.lndHarness.Alice.RPC.NewAddress(&lnrpc.NewAddressRequest{
+		Type: lnrpc.AddressType_TAPROOT_PUBKEY,
+	})
+
+	aliceP2TR, err := btcutil.DecodeAddress(
+		addrResp.Address, harnessNetParams,
+	)
+	require.NoError(t.t, err)
+
+	alicePkScript, err := txscript.PayToAddrScript(aliceP2TR)
+	require.NoError(t.t, err)
+
+	// These are basically Alice's terms that she signs the assets over:
+	// Send me 69420 satoshis to this address that belongs to me, and you
+	// will get assets in return.
+	btcpsbt.UnsignedTx.TxOut[0].PkScript = alicePkScript
+	btcpsbt.UnsignedTx.TxOut[0].Value = 69420
+	derivation, trDerivation := getAddressBip32Derivation(
+		t.t, addrResp.Address, t.lndHarness.Alice,
+	)
+	btcpsbt.Outputs[0].Bip32Derivation = []*psbt.Bip32Derivation{
+		derivation,
+	}
+	btcpsbt.Outputs[0].TaprootBip32Derivation = []*psbt.TaprootBip32Derivation{
+		trDerivation,
+	}
+	btcpsbt.Outputs[0].TaprootInternalKey = trDerivation.XOnlyPubKey
+
+	var b bytes.Buffer
+	err = btcpsbt.Serialize(&b)
+	require.NoError(t.t, err)
+
+	// Now we need to commit the vPSBT and PSBT, creating all the related
+	// proofs for this transfer to be valid.
+	resp, err := alice.CommitVirtualPsbts(
+		ctxb, &wrpc.CommitVirtualPsbtsRequest{
+			VirtualPsbts: [][]byte{signedResp.SignedPsbt},
+			AnchorPsbt:   b.Bytes(),
+			AnchorChangeOutput: &wrpc.CommitVirtualPsbtsRequest_Add{
+				Add: true,
+			},
+			Fees: &wrpc.CommitVirtualPsbtsRequest_TargetConf{
+				TargetConf: 12,
+			},
+		},
+	)
+	require.NoError(t.t, err)
+
+	// Now we retrieve the bitcoin PSBT from the response.
+	btcpsbt, err = psbt.NewFromRawBytes(
+		bytes.NewReader(resp.AnchorPsbt), false,
+	)
+	require.NoError(t.t, err)
+
+	// The first input is the anchor that carries Alice's assets. This input
+	// will only commit to itself and Alice's output at same index (0), which
```

Comment:
Maybe: "This input is committed to the first output" rather than "This input
	// will only commit to itself and Alice's output"
	
I think that we "commit an input to something" is correct rather than "the input commits itself"? I'm not sure how "commit" is commonly used in this context.

Reply:
Hmm not sure about the exact phrasing here. But the comment should definitely capture the fact that the input actually commits to "itself", as it is valid only if it's present in the final tx and also at the same tx as the output it was signing.

Reply:
I understand it this way: The input itself does not commit to anything, as it is not a signature. Rather, it is the signature generated by Alice that commits to the input.

I think you're also saying that the signature generated by Alice actually commits to more than just the input. It commits to the input and also the output at index 0. And that the final tx is only valid if it includes this input and output.

Reply:
> I understand it this way: The input itself does not commit to anything, as it is not a signature. Rather, it is the signature generated by Alice that commits to the input.

Right, my phrasing is wrong there, it's the signature that commits not the input.

> I think you're also saying that the signature generated by Alice actually commits to more than just the input. It commits to the input and also the output at index 0. And that the final tx is only valid if it includes this input and output.

Yeah this input's signature will commit to that input and the output at the same index (`0`). The fact that this input & output need to be in the tx is more like stating the obvious, could just strip that part